src/tracer.js

```diff
@@ -209,7 +209,7 @@ export default class Tracer {
         let internalTags: any = {};
         if (!parent || !parent.isValid) {
```

Comment:
I think this could be refactored to avoid repetition. For example, the invalid parent behavior can be done upfront:

```
if (parent && !parent.isValid()) {
   // ...
   parent = null;
}
```

then create new Context and update it depending on parent == null

then check if context is deferred and make sampling decision